### ISSUE TITLE
a11y test failures

### ISSUE THREAD

**Issue by bennypowers** [RedHat-UX/red-hat-design-system#582]
chai-a11y-axe fails for:
- rh-footer
- rh-tooltip
- rh-stat

see commits 62983d10a07bd1cd3bfdd6d9fe0e8a51035c5422, 8bd032fc14e3d6fe622000f3d53a52066fff9c6b, and 6291db74d1347c261184b124a2e10a7c4e3c1a0a

These are probably true failures, meaning elements should probably be fixed. Tooltip in particular will need to be fixed upstream in pfe (when the tooltip is not shown, the #invoker element is `aria-labelledby` #tooltip, which is aria-hidden at that time).

As well, some guidelines on writing tests should be added to the wiki, including elaborations on:
- using nested `describe`, `beforeEach(async function() {})`, avoiding `it(async`, and `this` context
- returning `.to.be.accessible()` (it's a Promise), or wrapping it in `await Promise.resolve()`. NB this was solved upstream but we haven't upgraded yet.
- writing fixture factories
- interaction tests
- what to test upstream, what to test downstream

**Comment by brianferry** [comment]
When updating `stat` to use the await promise 

```
it('passes the a11y audit', async function() {
    await Promise.resolve(expect(element).to.be.accessible());
});
```

There do not appear to be any errors on my end (running `npm run test elements/rh-stat/test/rh-stat.spec.ts`)

`tooltip` appears to have a problem with duplicate `ids` which I have a fix for though I'm not seeing the labelledby error anymore

**Comment by zeroedin** [comment]
> returning .to.be.accessible() (it's a Promise), or wrapping it in await Promise.resolve(). NB this was solved upstream but we haven't upgraded yet.

@bennypowers in taking a look at this to gain more understanding of what are you suggesting. How are the two below different?  Both seemingly pass. 

```
    it('passes the a11y audit', async function() {
      await expect(element).to.be.accessible();
    });
```
vs:

```
    it('passes the a11y audit', async function() {
      await Promise.resolve(expect(element).to.be.accessible());
    });
```

Are you saying the former allows the potential of a false positive pass?

/cc @brianferry
